File: go/mbpay/pay_order_info.go
package mbpay

import (
	"fmt"
	"net/http"
	"strconv"
)

// GetPayOrderInfo 查询支付订单信息
func (c *Client) GetPayOrderInfo(orderNo string, merchantID int64) (*PayOrderInfoResponse, error) {
	// 参数验证
	if orderNo == "" {
		return nil, fmt.Errorf("order_no is required")
	}
	if merchantID <= 0 {
		return nil, fmt.Errorf("merchant_id is required and must be greater than 0")
	}

	// 构建请求参数
	params := map[string]string{
		"order_no":    orderNo,
		"merchant_id": strconv.FormatInt(merchantID, 10),
	}

	// 执行请求
	resp, err := c.doRequest(http.MethodPost, "/merchant/payorderinfo", params)
	if err != nil {
		return nil, err
	}

	// 检查业务错误
	if !resp.IsSuccess() {
		return nil, resp.ToError()
	}

	// 解析 data 字段
	payOrderResp := &PayOrderInfoResponse{}

	if orderNo, ok := resp.Data["order_no"].(string); ok {
		payOrderResp.OrderNo = orderNo
	} else {
		return nil, fmt.Errorf("invalid order_no format in response")
	}

	if platformOrderNo, ok := resp.Data["platform_order_no"].(string); ok {
		payOrderResp.PlatformOrderNo = platformOrderNo
	} else {
		payOrderResp.PlatformOrderNo = "" // 可能为空
	}

	if amount, ok := resp.Data["amount"].(float64); ok {
		payOrderResp.Amount = int64(amount)
	} else if amount, ok := resp.Data["amount"].(int64); ok {
		payOrderResp.Amount = amount
	} else if amount, ok := resp.Data["amount"].(int); ok {
		payOrderResp.Amount = int64(amount)
	} else {
		return nil, fmt.Errorf("invalid amount format in response")
	}

	if fee, ok := resp.Data["fee"].(float64); ok {
		payOrderResp.Fee = int64(fee)
	} else if fee, ok := resp.Data["fee"].(int64); ok {
		payOrderResp.Fee = fee
	} else if fee, ok := resp.Data["fee"].(int); ok {
		payOrderResp.Fee = int64(fee)
	} else {
		payOrderResp.Fee = 0 // 默认为 0
	}

	if actualAmount, ok := resp.Data["actual_amount"].(float64); ok {
		payOrderResp.ActualAmount = int64(actualAmount)
	} else if actualAmount, ok := resp.Data["actual_amount"].(int64); ok {
		payOrderResp.ActualAmount = actualAmount
	} else if actualAmount, ok := resp.Data["actual_amount"].(int); ok {
		payOrderResp.ActualAmount = int64(actualAmount)
	} else {
		payOrderResp.ActualAmount = 0 // 默认为 0
	}

	if status, ok := resp.Data["status"].(float64); ok {
		payOrderResp.Status = int(status)
	} else if status, ok := resp.Data["status"].(int); ok {
		payOrderResp.Status = status
	} else {
		return nil, fmt.Errorf("invalid status format in response")
	}

	if statusText, ok := resp.Data["status_text"].(string); ok {
		payOrderResp.StatusText = statusText
	} else {
		payOrderResp.StatusText = "" // 可能为空
	}

	if remark, ok := resp.Data["remark"].(string); ok {
		payOrderResp.Remark = remark
	} else {
		payOrderResp.Remark = "" // 可能为空
	}

	if createAt, ok := resp.Data["create_at"].(string); ok {
		payOrderResp.CreateAt = createAt
	} else {
		payOrderResp.CreateAt = "" // 可能为空
	}

	if updateAt, ok := resp.Data["update_at"].(string); ok {
		payOrderResp.UpdateAt = updateAt
	} else {
		payOrderResp.UpdateAt = "" // 可能为空
	}

	if payAddress, ok := resp.Data["pay_address"].(string); ok {
		payOrderResp.PayAddress = payAddress
	} else {
		payOrderResp.PayAddress = "" // 可能为空
	}

	return payOrderResp, nil
}

File: python/mbpay/pay_order_info.py
"""支付订单信息查询功能"""

from .client import Client
from .types import PayOrderInfoResponse, Response, MBPayError


def get_pay_order_info(self: Client, order_no: str, merchant_id: int) -> PayOrderInfoResponse:
    """
    查询支付订单信息
    
    Args:
        order_no: 商户订单号（必填）
        merchant_id: 商户ID（必填）
        
    Returns:
        PayOrderInfoResponse: 支付订单信息响应
        
    Raises:
        MBPayError: API 业务错误
        requests.RequestException: 网络请求错误
    """
    # 参数验证
    if not order_no:
        raise MBPayError(0, "order_no is required")
    if merchant_id <= 0:
        raise MBPayError(0, "merchant_id is required and must be greater than 0")
    
    # 构建请求参数
    params = {
        "order_no": order_no,
        "merchant_id": str(merchant_id),
    }
    
    # 执行请求
    resp: Response = self._do_request("POST", "/merchant/payorderinfo", params)
    
    # 解析 data 字段
    data = resp.data
    order_no_resp = data.get("order_no", "")
    platform_order_no = data.get("platform_order_no", "")
    amount = int(data.get("amount", 0))
    fee = int(data.get("fee", 0))
    actual_amount = int(data.get("actual_amount", 0))
    status = int(data.get("status", 0))
    status_text = data.get("status_text", "")
    remark = data.get("remark", "")
    create_at = data.get("create_at", "")
    update_at = data.get("update_at", "")
    pay_address = data.get("pay_address", "")
    
    if not order_no_resp:
        raise MBPayError(0, "invalid order_no format in response")
    
    return PayOrderInfoResponse(
        order_no=order_no_resp,
        platform_order_no=platform_order_no,
        amount=amount,
        fee=fee,
        actual_amount=actual_amount,
        status=status,
        status_text=status_text,
        remark=remark,
        create_at=create_at,
        update_at=update_at,
        pay_address=pay_address
    )


# 将方法绑定到 Client 类
Client.get_pay_order_info = get_pay_order_info

